### PR TITLE
Revamping for Xcode 9, Swift 4 migration, Pods update

### LINKED ISSUE #227: Revamping for xcode 9
**Actual Behaviour**

App doesn't run and gives an error of 'Ambiguous use of title' on line 22 of TrainingVCMethods.swift
App doesn't run and gives an error of 'Ambiguous use of title' on line 20 of ResetPasswordVCMethods.swift
Note: This issue now includes all the occurences of navigationItem.title = ... and they all have now been changed to navigationItem.titleLabel.text.

**Expected Behaviour**

App was supposed to run, guess it was a bug in xcode 9. But i have fixed it and i'll be building for ios 11.

**Steps to reproduce it**

Replace titleLabel with title and you get the error

**Would you like to work on the issue?**
Yes, gladly

### PR DESCRIPTION
**Pull request by imjog**
Fixes #227 

Changes:
* **Updated pods** - Pods were old swift 3 compatible so updated pods to the latest version for Swift 4.
* **Swift 4 Migration** - Migrated from Swift 3 to latest Swift 4, Write swift 4 syntaxes, resolved compiler warnings, updated deprecated code.

Screenshots for the change: 
Resolved all the errors and compiler warnings -
screen shot 2018-03-06 at 16 51 20

### CODE REVIEW

**Comment by imjog** [comment]
I think Circle Ci is failing because of old xcode version defined in `circle.yml` file.

**Comment by imjog** [comment]
Hi @chashmeetsingh  I resolved all the realm pod self-retain warnings in xcode but CircleCi is still showing that warnings. Also `AuthenticationUITests.testLogoutSuccess()` test is failing, I think this is because of network calling problem. Please look into it. Thanks

File: Susi/Controllers/ChatViewController/ChatVCMethods.swift [hunk 1]
```diff
@@ -343,12 +343,17 @@ extension ChatViewController {
...
         let size = CGSize(width: 250, height: 1000)
         let options = NSStringDrawingOptions.usesFontLeading.union(.usesLineFragmentOrigin)
-        return NSString(string: message).boundingRect(with: size, options: options, attributes: [NSFontAttributeName: UIFont.systemFont(ofSize: 16)], context: nil)
+        return NSString(string: message).boundingRect(with: size, options: options, attributes: [NSAttributedStringKey.font: UIFont.systemFont(ofSize: 16)], context: nil)
     }
 
     // loads all messages from database
     func loadMessages() {
-        messages = List<Message>(realm.objects(Message.self))
+        let results: Results<Message> = realm.objects(Message.self)
```
Comment (chashmeetsingh) [inline Susi/Controllers/ChatViewController/ChatVCMethods.swift]: No need to convert here when the `Results<Message>` object can be used. Instead change the type of messages to this.

File: Susi/Controllers/ChatViewController/SpeechRecognizerDelegate.swift [hunk 2]
```diff
@@ -67,9 +67,7 @@ extension ChatViewController: SFSpeechRecognizerDelegate, AVSpeechSynthesizerDel
 
         recognitionRequest = SFSpeechAudioBufferRecognitionRequest()
 
-        guard let inputNode = audioEngine.inputNode else {
-            fatalError("Audio engine has no input node")
-        }
+        let inputNode = audioEngine.inputNode
```
Comment (chashmeetsingh) [inline Susi/Controllers/ChatViewController/SpeechRecognizerDelegate.swift]: Any specific reason for this change?

File: Susi/Controllers/TrainingViewController/SpeechRecognitionDelegate.swift [hunk 3]
```diff
@@ -56,9 +56,10 @@ extension TrainingViewController: SFSpeechRecognizerDelegate {
 
         recognitionRequest = SFSpeechAudioBufferRecognitionRequest()
 
-        guard let inputNode = audioEngine.inputNode else {
-            fatalError("Audio engine has no input node")
-        }
+//        guard let inputNode = audioEngine.inputNode else {
+//            fatalError("Audio engine has no input node")
+//        }
+        let inputNode = audioEngine.inputNode
```
Comment (chashmeetsingh) [inline Susi/Controllers/TrainingViewController/SpeechRecognitionDelegate.swift]: Same here.

File: Susi/Controllers/TrainingViewController/TrainingViewController.swift [hunk 4]
```diff
@@ -41,7 +41,7 @@ class TrainingViewController: GeneralViewController {
     @IBOutlet weak var recordPhI3: UIImageView!
     @IBOutlet weak var finishLaterButton: FlatButton!
     @IBOutlet weak var downloadIndicator: UIActivityIndicatorView!
-    @IBOutlet weak var downloadLabel: UILabel!
+    @objc @IBOutlet weak var downloadLabel: UILabel!
```
Comment (chashmeetsingh) [inline Susi/Controllers/TrainingViewController/TrainingViewController.swift]: Why is @objc required here?

File: Susi/Controllers/ChatViewController/SpeechRecognizerDelegate.swift (hunk 2 above)
Comment (imjog) [reply Susi/Controllers/ChatViewController/SpeechRecognizerDelegate.swift]: `inputNode` is defined as let `audioEngine = AVAudioEngine()` which is not optional. when we use guard let it is giving obvious error: 
> Initializer for conditional binding must have Optional type, not 'AVAudioInputNode'

File: Susi/Controllers/ChatViewController/ChatVCMethods.swift (hunk 1 above)
Comment (imjog) [reply Susi/Controllers/ChatViewController/ChatVCMethods.swift]: Okey! Now I declared `messages` as `var messages = [Message]()` and used `messages = Array(realm.objects(Message.self))` in the `loadMessages` function.

**Pull request merged**